test/parallel/test-timers-invalid-clear.js

```
@@ -0,0 +1,31 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+
+async function test() {
+  const t = setTimeout(() => {
+    console.log('1');
+  }, 0);
+
+  clearImmediate(t);
```

hmm... I wonder if this case shouldn't actually throw an error rather than just be silently ignored. It's obviously incorrect to use `clearImmediate(...)` with a timeout.

---

It's not a standard so I couldn't know what to do, we can emit a warning maybe?

---

Or I'm down to throwing too

---

Others only check for the property, so I think in terms of alignment, starting with this is good

---

FWIW both `const t = setTimeout(function() {console.log(1)}, 1000); clearInterval(t);` and `const t = setInterval(function() {console.log(1)}, 1000); clearTimeout(t);` work in the browser.